src/context/CartContext.jsx

```diff
@@ -0,0 +1,143 @@
+import axios from "axios";
+import React, { useState, createContext, useContext, useEffect } from "react";
+import { useAuth } from "./AuthContext";
+import { useToast } from "./ToastContext";
+const CartContext = createContext();
+const initialState = {
+  loading: true,
+  data: [],
+  error: "",
+};
+const CartProvider = ({ children }) => {
+  const { setToast } = useToast();
+  const { isLoggedIn } = useAuth();
+  const [cart, setCart] = useState(initialState);
+  const addToCart = (product) => {
+    if (!isLoggedIn) {
+      return;
+    }
+    if (cart.data.find((item) => item._id === product._id)) {
+      setToast({
+        show: true,
+        content: `Item already added to the cart`,
+        type: "info",
+      });
+    } else {
+      axios
+        .post(
+          "/api/user/cart",
+          JSON.stringify({
+            product,
+          }),
+          {
+            headers: { authorization: localStorage.getItem("token") },
```

Comment:
Just a suggestion - You can save the token the in Auth Context Varibale, so that you don't have to get it from localStorage. Rest everyhting looks good to merge

Reply:
Yes, I can get it from authcontext

Reply:
Fixed